Misc/NEWS.d/next/Core and Builtins/2021-08-09-19-05-20.bpo-44874.oOcfU4.rst

```diff
@@ -0,0 +1,38 @@
+Deprecate the old trashcan macros (``Py_TRASHCAN_SAFE_BEGIN``/``Py_TRASHCAN_SAFE_END``). They should be replaced by the new macros ``Py_TRASHCAN_BEGIN`` and ``Py_TRASHCAN_END``.
+
+A tp_dealloc function that has the old macros, such as::
+
+   static void
+   mytype_dealloc(mytype *p)
+   {
+       PyObject_GC_UnTrack(p);
+       Py_TRASHCAN_SAFE_BEGIN(p);
+       ...
+       Py_TRASHCAN_SAFE_END
+   }
+
+should migrate to the new macros as follows::
+
+   static void
+   mytype_dealloc(mytype *p)
+   {
+       PyObject_GC_UnTrack(p);
+       Py_TRASHCAN_BEGIN(p, mytype_dealloc)
+       ...
+       Py_TRASHCAN_END
+   }
+
+Note that ``Py_TRASHCAN_BEGIN`` has a second argument which
+ should be the deallocation function it is in.
+
+To support older python versions in the same codebase, you
+can define the following macros and use them throughout
+the code (credit: these were copied from the ``mypy`` codebase)::
```

Comment:
It seems like mypy is released under the MIT license: https://github.com/python/mypy/blob/master/LICENSE

Reply:
How does this impact the example?